src/Map.cpp

```diff
@@ -65,22 +65,22 @@ void cMap::Tick()
 
 
 
-void cMap::UpdateRadius(int a_PixelX, int a_PixelZ, unsigned int a_Radius)
+void cMap::UpdateRadius(unsigned int a_PixelX, unsigned int a_PixelZ, unsigned int a_Radius)
 {
-	int PixelRadius = a_Radius / GetPixelWidth();
+	unsigned int PixelRadius = a_Radius / GetPixelWidth();
 
-	unsigned int StartX = Clamp(a_PixelX - PixelRadius, 0, (int)m_Width);
-	unsigned int StartZ = Clamp(a_PixelZ - PixelRadius, 0, (int)m_Height);
+	unsigned int StartX = Clamp(a_PixelX - PixelRadius, static_cast<unsigned int>(0), m_Width);
+	unsigned int StartZ = Clamp(a_PixelZ - PixelRadius, static_cast<unsigned int>(0), m_Height);
```

Comment:
This might not work well - it is causing `Clamp<unsigned int>` to be used, therefore forcing `a_PixelX - PixelRadius` into an unsigned int before passing it to `Clamp`. It needs to use `Clamp<int>` and then cast it into an unsigned, if needed (why can't all of them just use ints and be done with it?)